src/Workspaces/Remote/ServiceHub/Services/ProcessTelemetry/RemoteWorkspaceTelemetryService.cs

```diff
@@ -6,24 +6,37 @@
 using System.Composition;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.Internal.Log;
-using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Telemetry;
 using Microsoft.VisualStudio.Telemetry;
 
 namespace Microsoft.VisualStudio.LanguageServices.Telemetry
 {
     [ExportWorkspaceService(typeof(IWorkspaceTelemetryService)), Shared]
-    internal sealed class RemoteWorkspaceTelemetryService : AbstractWorkspaceTelemetryService
+    internal sealed class RemoteWorkspaceTelemetryService : AbstractWorkspaceTelemetryService, IDisposable
     {
+        private TelemetryLogger? _telemetryLogger;
```

Comment:
Just to confirm: The fact that the new telemetry logger is only hooked up in the remote workspace and not host workspace is not a limitation of the logger but just the need for this PR, correct?

Reply:
I'm not clear on this question, can you clarify? TelemetryLogger.Create is called both from RemoteWorkspaceTelemetryService and from VisualStudioWorkspaceTelemetryService.